### PR TITLE
Switch to tensorboard.dev for training metadata handling

### PR DESCRIPTION
**Pull request by ianjjohnson**
This changes update_training_history.py to upload tensorboard logs to tensorboard.dev, and then use tensorboard.dev's API to load relevant metrics from training logs.

This also switches the script to support both flags and CLI input to give the user some flexibility.

@LukeWood

### CODE REVIEW

File: shell/weights/update_training_history.py [hunk 1]
```diff
@@ -1,54 +1,92 @@
...
+    "weights_version",
     None,
-    "Path to the JSON file for run history for the producing training script",
+    "The version of the training script used to produce the latest weights. For example, v0",
+)
+flags.DEFINE_string(
+    "pr_number", None, "The PR number that merged the training script into KerasCV"
+)
+flags.DEFINE_string(
+    "pr_author", None, "The GitHub username of the author of the training script"
```
Comment (LukeWood) [inline shell/weights/update_training_history.py:19]: looks like this isn't inserted into the metadata anywhere

File: shell/weights/update_training_history.py [hunk 2]
```diff
@@ -1,54 +1,92 @@
...
-# Find the single validation logs in the supplied logs dir
-files = os.listdir(FLAGS.logs_path)
-# Tensorboard logs are named based on timestamp, so the largest name alphabetically is the most recent.
-most_recent_file = max(files)
+weights_version = FLAGS.weights_version or input(
+    "Input the weights version for your script\n"
+)
 
-filename = FLAGS.logs_path + "/" + most_recent_file
+training_script_path = FLAGS.training_script_path or input(
```
Comment (LukeWood) [inline shell/weights/update_training_history.py:29]: should we also allow an input for author?

File: shell/weights/update_training_history.py [hunk 3]
```diff
@@ -1,54 +1,92 @@
...
-print(f"Max validation accuracy: {max_validation_accuracy}")
+new_results = {
+    "validation_accuracy": max_validation_accuracy,
+    "epochs_trained": training_epochs,
+    "tensorboard_logs": f"https://tensorboard.dev/experiment/{tensorboard_experiment_id}/",
+}
 
-new_results = {FLAGS.weights_version: {"validation_accuracy": max_validation_accuracy}}
+pr_number = FLAGS.pr_number or input(
+    "Input the PR number that merged your script into KerasCV\n"
```
Comment (LukeWood) [inline shell/weights/update_training_history.py:77]: should this be a link instead of a number?

File: shell/weights/update_training_history.py [hunk 4]
```diff
@@ -1,54 +1,92 @@
...
+# Check if the JSON file already exists
+results_file = open(training_script_json_path, "r")
 results_string = results_file.read()
 results = json.loads(results_string) if results_string != "" else {}
 results_file.close()
 
-results_file = open(FLAGS.training_script_json, "w")
-results[FLAGS.weights_version] = {"validation_accuracy": max_validation_accuracy}
+results_file = open(training_script_json_path, "w")
+results[weights_version] = new_results
```
Comment (LukeWood) [inline shell/weights/update_training_history.py:90]: can we write the PR result to the results file too?

**Review by LukeWood — CHANGES_REQUESTED** [review]

**Comment by LukeWood** [comment]
Thanks for the PR!  A few updates

File: shell/weights/update_training_history.py (hunk 3 above)
Comment (ianjjohnson) [reply shell/weights/update_training_history.py:77]: Good call -- I've kept the flag as pr_number but the user input version is a full link.

Comment (LukeWood) [reply shell/weights/update_training_history.py]: seems reasonable

**Review by LukeWood — APPROVED** [review]

**Comment by LukeWood** [comment]
Cool, great use of the integrations

**Pull request merged**